### ISSUE TITLE
can't find Prepare_TestData_HR_LR.m for test and some problem

### ISSUE THREAD

**Issue by stefenmax** [MIVRC/MLEFGN-PyTorch#1]
i've been successfully training in your default setting,but when i want to test,i can't find the .m doc.

And interestingly,i find the WGAN in the loss function,but i don't see it in u network,does it means even without the generator one can still use the adversial loss?

thank u very much.wonderful work

**Comment by CV-JunchengLi** [comment]
Thank you for your attention.
I found that I did miss the file, I will upload it as soon as possible (including all pre-trained model).
As for the adversarial loss, MLEFGN is a CNN-based model, not a GAN-based model. Therefore, we do not use adversarial loss in our model. You find the WGAN in the loss function means that if need you can use it, but it not need in MLEFGN.

**Comment by stefenmax** [comment]
Thanks for your quick response !
By the way.i found the testdata.m in your another project **seaNet**,but i don't know how to use,could you tell me how it works in the test data briefly?Because i tried test mode in my trained model without process in MATLAB,it still works and not bad.

And i also want to use your modle in gray image,when i tried in my own gray dataset and setting --n_colors =1,it raise me an error like that
![image](https://user-images.githubusercontent.com/45379408/92211705-70663000-eec3-11ea-9a99-3b6739a42614.png)
Does i need another options to set? 
Thanks!!

**Comment by CV-JunchengLi** [comment]
1、I checked the code and found that this is an error in the readme file. This is an image denoising task. Therefore, Prepare_TestData_HR_LR.m is not needed. If you want to generate noisy images for the test, you can use the code "noise_image = imnoise(im_hr,'gaussian',0,(noise_std/255)^2);" in https://github.com/MIVRC/MLEFGN-PyTorch/blob/master/Matlab/generate_noise.m.

2、The Prepare_TestData_HR_LR.m in the SeaNet project is used to generate low-resolution images, which is not suitable for this image denoising task.

3、You can try setting --n_colors = 3. In the code, if your image is a gray image (on one channel), the code will duplicate it twice to become a three channels image. In other words, the contents of the three channels are consistent.

def set_channel(*args, n_channels=3):
    def _set_channel(img):
        if img.ndim == 2:
            img = np.expand_dims(img, axis=2)

        c = img.shape[2]
        if n_channels == 1 and c == 3:
            img = np.expand_dims(sc.rgb2ycbcr(img)[:, :, 0], 2)
        elif n_channels == 3 and c == 1:
            img = np.concatenate([img] * n_channels, 2)

        return img

    return [_set_channel(a) for a in args]

**Comment by stefenmax** [comment]
Thanks,follow your instruction i finall pass the code in gray image with little modify in your original code.
When i debug my dataset in the coda,i find that the bug rasie before the function set_channel used,so i switch the code between https://github.com/MIVRC/MLEFGN-PyTorch/blob/master/Train/data/srdata.py#L174 and L175.
And it finally works.
Please check it.
Thanks a lot.

**Comment by CV-JunchengLi** [comment]
Thanks for your suggestions, I will check and update the code after the new paper‘s ddl.
